File: app/src/main/java/nejati/me/bliss/base/BaseViewModel.java
package nejati.me.bliss.base;



import androidx.lifecycle.ViewModel;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BaseViewModel<N> extends ViewModel
{
    private N mNavigator;

    private CompositeDisposable mCompositeDisposable;

    public BaseViewModel()
    {
        this.mCompositeDisposable = new CompositeDisposable();
    }

    @Override
    protected void onCleared()
    {
        mCompositeDisposable.dispose();
        super.onCleared();
    }

    public CompositeDisposable getCompositeDisposable()
    {
        return mCompositeDisposable;
    }

    public N getNavigator()
    {
        return mNavigator;
    }

    public void setNavigator(N mNavigator)
    {
        this.mNavigator = mNavigator;
    }

//    public void setIsLoading(boolean isLoading)
//    {
//        this.isLoading.set(isLoading);
//    }
}

File: app/src/main/java/nejati/me/bliss/base/BaseViewModel.kt
package nejati.me.bliss.base


import androidx.databinding.ObservableField
import androidx.lifecycle.ViewModel

import io.reactivex.disposables.CompositeDisposable

/**
 * @param N Activity Navigator
 */
abstract class BaseViewModel<N> : ViewModel() {

    var showRertryLayout = ObservableField<Boolean>()
    var showNoInternetLayout = ObservableField<Boolean>()

    var navigator: N? = null

    /**
     * Retry Call Api Widget
     */
    abstract fun OnClickRetryAction()

    /**
     * Monitoriing Internet Connection
     * @param status true  is Internet available, false is Internet not available
     */
    abstract fun isInternetAvilable(status:Boolean)

    val compositeDisposable: CompositeDisposable

    init {
        this.compositeDisposable = CompositeDisposable()
    }

    override fun onCleared() {
        compositeDisposable.dispose()
        super.onCleared()

    }
}
